### DISCUSSION TITLE
Improve application configuration's documentation

### DISCUSSION THREAD

**Discussion by jorgecarleitao** [fastapi/fastapi#8161] [Questions] [answered]
**Is your feature request related to a problem? Please describe.**
When a user goes to the documentation on [configuration](https://fastapi.tiangolo.com/tutorial/application-configuration/), the first thing that he likely expects is how to configure the application from a deployment's point of view (e.g. SECRET_KEY, DB url, etc), e.g. see [starlette's page](https://www.starlette.io/config/).

**Describe the solution you'd like**
It would be interesting to have, as part of the documentation for theconfiguration, something about the deployment configuration (as defined above). If FastAPI does not have any requirement, then it could be as simple as

> FastAPI is based on Starlette, and thus you can configure it in the same was as a starlette application, see here for details.

An important part of this aspect, and one that users likely need help with, is how to setup the configuration on a large application with `APIRouter`s. Two examples:

* How to initialize the application depending on the configuration? For users coming from Flask, this would be the `create_app` part of it.

* How to access configuration values from routes in a module declaring routes for an APIRouter? Should the configuration be a `Depend`? For users coming from Flask, they are easily accessible through `flask.current_app.config`, something that does not exist in FastAPI.

**Comment by euri10** [comment]
the gitter chat had a pretty interesting discussion on that, might be worth checking from May 30 09:54 (sorry cant make a link apparently)

**Comment by jekirl** [comment]
For settings/config (both in general and with fastapi), I am fan of using `BaseSettings` from `pydantic` over starlette's config class

**Comment by tiangolo** [comment, marked as answer]
Yes, we should have a section in the docs about using Pydantic's `BaseSettings`.

Let's track the progress on this issue: https://github.com/tiangolo/fastapi/issues/190

**Reply by Kludex** [reply]
This is now documented on https://fastapi.tiangolo.com/advanced/settings/?h=basesettings#create-the-settings-object. 🙏
